ksqldb-engine/src/main/java/io/confluent/ksql/security/DefaultKsqlPrincipal.java

```diff
@@ -28,9 +28,15 @@
 public class DefaultKsqlPrincipal implements KsqlPrincipal {
 
   private final Principal principal;
+  private String ipAddress;
```

Comment:
> ipAddress can be final.

Good point. Fixed!

> Since the ip address is an optional parameter, should we wrap it into Optional?

The vert.x docs say it's only null in the case of domain sockets, which would be a very odd usage pattern for sending requests to a ksqlDB server. That said, we should definitely handle the null case to avoid NPEs (great catch). I don't think it's worth wrapping this into an Optional to handle an uncommon usage pattern. I've left it unwrapped but did add the null check into DefaultApiSecurityContext as suggested. LMK what you think.

Reply:
My main reason to suggest `Optional` was to avoid passing an empty string as a "special" value. In other words, if `ipAddress` is present, then it is always valid.

I don't think that it is too important and should block the PR, so feel free to merge.